docs/csharp/pattern-matching.md

```diff
@@ -1,268 +1,70 @@
-readable, concise code. Pattern matching enables idioms where data and
-the code are separated, unlike object-oriented designs where data
-and the methods that manipulate them are tightly coupled.
+This article provides an overview of scenarios where you can use pattern matching. These techniques can improve the readability and correctness of your code. For a full discussion of all the patterns you can apply, see the article on [patterns](language-reference/operators/patterns.md) in the language reference.
 
-To illustrate these new idioms, let's work with structures that represent
-geometric shapes using pattern matching statements. You're probably
-familiar with building class hierarchies and creating
-[virtual methods and overridden methods](methods.md#inherited) to
-customize object behavior based on the runtime type of the object.
+## Null checks
 
-Those techniques aren't possible for data that isn't structured in a class
-hierarchy. When data and methods are separate, you need other tools. The new
-*pattern matching* constructs enable cleaner syntax to examine data
-and manipulate control flow based on any condition of that data. You already
-write `if` statements and `switch` that test a variable's value. You write `is`
-statements that test a variable's type. *Pattern matching* adds new capabilities
-to those statements.
+One of the most common scenarios for pattern matching is to ensure values aren't null. You can test and convert a nullable value type to its underlying type while testing for `null` using the following example:
 
-In this article, you'll build a method  that computes the area of
-different geometric shapes. But, you'll do it without resorting to object-oriented techniques and building a class hierarchy for the different shapes.
-You'll use *pattern matching* instead.
-As you go through this sample, contrast this code with how it would
-be structured as an object hierarchy. When the data you must
-query and manipulate isn't a class hierarchy, pattern matching enables
-elegant designs.
+:::code language="csharp" source="snippets/patterns/Program.cs" ID="NullableCheck":::
 
-Rather than starting with an abstract shape definition and adding different
-specific shape classes, let's start instead with simple data only definitions
-for each of the geometric shapes:
+The preceding code is a [*declaration pattern*](language-reference/operators/patterns.md#declaration-and-type-patterns) to test the type of the variable, and assign it to a new variable. The language rules make this technique safer than many others. The variable `number` is only accessible and assigned in the true portion of the `if` clause. If you try to access it elsewhere, either in the `else` clause, or after the `if` block, the compiler issues an error. Secondly, because you're not using the `==` operator, this pattern works when a type has overridden the `==` operator. That makes it an ideal way to check null reference values, adding the `not` pattern:
```

Comment:
```suggestion
The preceding code is a [*declaration pattern*](language-reference/operators/patterns.md#declaration-and-type-patterns) to test the type of the variable, and assign it to a new variable. The language rules make this technique safer than many others. The variable `number` is only accessible and assigned in the true portion of the `if` clause. If you try to access it elsewhere, either in the `else` clause, or after the `if` block, the compiler issues an error. Secondly, because you're not using the `==` operator, this pattern works when a type overloads the `==` operator. That makes it an ideal way to check null reference values, adding the `not` pattern:
```